pkg/controller/identity/iamrolepolicyattachment/controller.go

```diff
@@ -83,37 +86,30 @@ func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (manag
 		return nil, errors.Wrap(err, errClient)
 	}
 
-	return &external{c}, nil
+	return &external{c, conn.client}, nil
 }
 
 type external struct {
 	client iam.RolePolicyAttachmentClient
+	kube   client.Client
 }
 
 func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
-	cr, ok := mgd.(*v1alpha3.IAMRolePolicyAttachment)
+	cr, ok := mgd.(*v1beta1.IAMRolePolicyAttachment)
 	if !ok {
 		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
 
-	req := e.client.ListAttachedRolePoliciesRequest(&awsiam.ListAttachedRolePoliciesInput{
-		RoleName: aws.String(cr.Spec.RoleName),
-	})
-
-	observed, err := req.Send(ctx)
+	observed, err := e.client.ListAttachedRolePoliciesRequest(&awsiam.ListAttachedRolePoliciesInput{
+		RoleName: aws.String(cr.Spec.ForProvider.RoleName),
+	}).Send(ctx)
 	if err != nil {
-		if iam.IsErrorNotFound(err) {
-			return managed.ExternalObservation{
-				ResourceExists: false,
-			}, nil
-		}
-
-		return managed.ExternalObservation{}, errors.Wrapf(err, errGet, cr.Spec.RoleName)
+		return managed.ExternalObservation{}, errors.Wrapf(resource.Ignore(iam.IsErrorNotFound, err), errGet, cr.Spec.ForProvider.RoleName)
 	}
 
 	var attachedPolicyObject *awsiam.AttachedPolicy
 	for i := 0; i < len(observed.AttachedPolicies); i++ {
```

Comment:
```suggestion
	for _, policy := range observed.AttachedPolicies {
		if cr.Spec.ForProvider.PolicyARN == aws.StringValue(policy.PolicyArn) {
			attachedPolicyObject = &policy
			break
		}
	}
```

Nitpick: using `range` would probably be easier to read.